src/Ng/NugetServiceEndpoints.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Ng.Models;
+using NuGet.Services.Metadata.Catalog;
+
+namespace Ng
+{
+    /// <summary>
+    /// Container type for all the NuGet service endpoints.
+    /// </summary>
+    class NugetServiceEndpoints
+    {
+        /// <summary>
+        /// Creates a new NugetServiceUrl type using the service endpoints defined in the serviceIndexUrl provided.
+        /// </summary>
+        /// <param name="serviceIndexUrl">The service index endpoint which contains the definition of the NuGet services. e.g. https://api.nuget.org/v3/index.json</param>
+        public NugetServiceEndpoints(Uri serviceIndexUrl)
+        {
+            this.InitializeAsync(serviceIndexUrl).Wait();
+        }
+
+        /// <summary>
+        /// The root service endpoint which contains the defintion for all the other endpoints. e.g. https://api.nuget.org/v3/index.json
+        /// </summary>
+        public Uri ServiceIndexUrl
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The base URL for the registration endpoints. e.g. https://api.nuget.org/v3/registration1/
+        /// </summary>
+        public Uri RegistrationBaseUrl
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Composes the registration URL for the specified package. e.g. https://api.nuget.org/v3/registration1/autofac.mvc2/2.3.2.632.json
+        /// </summary>
+        /// <param name="packageId">The package id.</param>
+        /// <param name="packageVersion">The package version.</param>
+        /// <returns>The registration URL for the specified package.</returns>
+        public Uri ComposeRegistrationUrl(string packageId, string packageVersion)
+        {
+            // The registration URL looks similar to https://api.nuget.org/v3/registration1/autofac.mvc2/2.3.2.632.json
+            string relativePath = String.Format("{0}/{1}.json", packageId, packageVersion);
```

Comment:
I had no idea that feature existed. :) Thanks for making me aware. I'll make the fix!